core/sr-primitives/src/generic/unchecked_mortal_compact_extrinsic.rs

```diff
@@ -137,17 +162,19 @@ where
 		Some(UncheckedMortalCompactExtrinsic {
 			signature: if is_signed { Some(Decode::decode(input)?) } else { None },
 			function: Decode::decode(input)?,
+			tip: if is_signed { Decode::decode(input).unwrap_or_default() } else { Tip::None },
```

Comment:
Why only handle `Tip` for signed transactions? Perhaps we could indeed bump the version and make it required?

Reply:
Agree that: 
- have to bump the version
- Now that tip is part of the signature payload it makes no sense to have it _optional_. It _must_ exist (no more `Decode::decode(input).unwrap_or_default()`, simply decode and fail if it is not there).